File: app/src/main/java/opencrowd/hgc/hgcwallet/ui/onboard/NewWalletCreateScreen.java
package opencrowd.hgc.hgcwallet.ui.onboard;

import android.content.Context;
import android.support.annotation.NonNull;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.wealthfront.magellan.BaseScreenView;
import com.wealthfront.magellan.Screen;

import java.util.List;

import opencrowd.hgc.hgcwallet.R;
import opencrowd.hgc.hgcwallet.common.Singleton;
import opencrowd.hgc.hgcwallet.common.UserSettings;
import opencrowd.hgc.hgcwallet.modals.HGCKeyType;
import opencrowd.hgc.hgcwallet.crypto.HGCSeed;
import opencrowd.hgc.hgcwallet.ui.customviews.TitleBarWrapper;


public class NewWalletCreateScreen extends Screen<NewWalletCreateView> {

    @NonNull
    HGCSeed mSeed;
    @NonNull HGCKeyType keyType;

    public NewWalletCreateScreen(@NonNull HGCKeyType keyType) {
        mSeed = Singleton.INSTANCE.createSeed();
        this.keyType = keyType;
    }

    @NonNull
    @Override
    protected NewWalletCreateView createView(@NonNull Context context) {
        return new NewWalletCreateView(context,mSeed,keyType);
    }
}
    class NewWalletCreateView extends BaseScreenView<NewWalletCreateScreen> {

        @NonNull List<String> mList;
        Button mCopy,mDone;
        private TextView mTextViewCryptoWords;

        public NewWalletCreateView(final @NonNull Context context, final @NonNull HGCSeed seed, final @NonNull HGCKeyType keyType) {
            super(context);
            inflate(context, R.layout.fragment_new_wallet, this);
            TitleBarWrapper titleBar = new TitleBarWrapper(findViewById(R.id.titleBar));
            titleBar.setTitle(getResources().getString(R.string.backup_your_wallet));
            mList = seed.toWordsList();

            mCopy = findViewById(R.id.btn_copy);
            mDone = findViewById(R.id.btn_done);
            mTextViewCryptoWords = findViewById(R.id.textview_crptowords);
            mTextViewCryptoWords.setText(TextUtils.join("   ",mList));

            titleBar.setCloseButtonHidden(false);
            titleBar.setOnCloseButtonClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                   getScreen().getNavigator().goBack();
                }
            });

            mCopy.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    Singleton.INSTANCE.copyToClipBoard(TextUtils.join(" ",mList),getContext());
                    Singleton.INSTANCE.showToast(getScreen().getActivity(), getScreen().getActivity().getString(R.string.copy_data_clipboard_passphrase));
                }
            });

            mDone.setOnClickListener(new OnClickListener() {
                @Override
                public void onClick(View view) {
                    getScreen().getNavigator().goTo(new PinSetUpOptionScreen(seed,keyType, null));
                }
            });
            UserSettings.instance.setValue(UserSettings.KEY_HAS_SHOWN_BIP39_MNEMONIC, true);
        }
    }

File: app/src/main/java/hedera/hgc/hgcwallet/ui/onboard/NewWalletCreateScreen.kt
/*
 *
 *  Copyright 2019 Hedera Hashgraph LLC
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */

package hedera.hgc.hgcwallet.ui.onboard

import android.content.Context
import android.text.TextUtils
import android.view.View
import android.widget.Button
import android.widget.TextView
import com.wealthfront.magellan.BaseScreenView
import com.wealthfront.magellan.Screen
import hedera.hgc.hgcwallet.R
import hedera.hgc.hgcwallet.common.Singleton
import hedera.hgc.hgcwallet.crypto.HGCSeed
import hedera.hgc.hgcwallet.modals.KeyDerivation
import hedera.hgc.hgcwallet.ui.customviews.TitleBarWrapper

class NewWalletCreateScreen() : Screen<NewWalletCreateView>() {

    data class Param(val seed: HGCSeed, val list: List<String>)

    private val param: Param

    init {
        val seed = Singleton.createSeed()
        val list = seed.toWordsList()
        param = Param(seed, list)
    }

    override fun createView(context: Context): NewWalletCreateView {
        return NewWalletCreateView(context, param)
    }

    internal fun onCloseButtonClick() {
        navigator?.goBack()
    }

    internal fun onCopyButtonClick() {
        Singleton.copyToClipBoard(TextUtils.join(" ", param.list), activity)
        Singleton.showToast(activity, activity.getString(R.string.copy_data_clipboard_passphrase))
    }

    internal fun onDoneButtonClick() {
        navigator?.goTo(PinSetUpOptionScreen(param.seed, KeyDerivation.BIP32, null));
    }
}

class NewWalletCreateView(context: Context, param: NewWalletCreateScreen.Param) : BaseScreenView<NewWalletCreateScreen>(context) {

    init {
        View.inflate(context, R.layout.fragment_new_wallet, this)
        TitleBarWrapper(findViewById(R.id.titleBar)).apply {
            setTitle(R.string.backup_your_wallet)
            setCloseButtonHidden(false)
            setOnCloseButtonClickListener {
                screen?.onCloseButtonClick()
            }
        }

        findViewById<TextView>(R.id.textview_crptowords)?.apply {
            text = TextUtils.join("   ", param.list)
        }


        findViewById<Button>(R.id.btn_copy)?.apply {
            setOnClickListener {
                screen?.onCopyButtonClick()
            }
        }

        findViewById<Button>(R.id.btn_done)?.apply {
            setOnClickListener {
                screen?.onDoneButtonClick()
            }
        }

    }
}